diff --git a/Android.bp b/Android.bp
index 745d2e8..8e44ff7 100644
--- a/Android.bp
+++ b/Android.bp
@@ -40,23 +40,30 @@ license {
 }
 
 rust_library {
     name: "libahash",
     host_supported: true,
     crate_name: "ahash",
     cargo_env_compat: true,
     cargo_pkg_version: "0.8.11",
     crate_root: "src/lib.rs",
     edition: "2018",
+    arch: {
+        arm64: {
+            features: ["folded_multiply"],
+        },
+        x86_64: {
+            features: ["folded_multiply"],
+        },
+    },
     features: [
         "default",
-        "folded_multiply",
         "getrandom",
         "runtime-rng",
         "specialize",
         "std",
     ],
     rustlibs: [
         "libcfg_if",
         "libgetrandom",
         "libonce_cell",
         "libzerocopy",
